velox/connectors/hive/HivePartitionFunction.cpp

```diff
@@ -29,19 +29,71 @@ void hashTyped(
       TypeTraits<kind>::name);
 }
 
-template <>
-void hashTyped<TypeKind::BOOLEAN>(
+template <typename T>
+void abstractHashTyped(
     const DecodedVector& values,
     vector_size_t size,
     bool mix,
+    std::function<uint32_t(const T&)> const& hashOne,
```

Comment:
can you use a function passed as an argument through a template argument, instead of std::function? std::function is a dynamic construct very unlikely to be inlined. should be avoided.
for this case, here those functions are applied per row. should be inlined, very simple ones.
```
template<typename Func>
void func(Func&& f){
f(); 
}
```

Reply:
Done, thanks for suggestion!